falcon/asgi/app.py

```diff
@@ -515,34 +563,114 @@ def __init__(self, *args, request_type=Request, response_type=Response, **kwargs
+        most specific one, as determined by the method resolution order of the
+        raised exception type. If multiple error handlers are registered for the
+        *same* exception class, then the most recently-registered handler is
+        used.
+
+        For example, suppose we register error handlers as follows::
+
+            app = App()
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_not_found)
+            app.add_error_handler(falcon.HTTPError, custom_handle_http_error)
+            app.add_error_handler(Exception, custom_handle_uncaught_exception)
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_404)
+
+        If an instance of ``falcon.HTTPForbidden`` is raised, it will be
+        handled by ``custom_handle_http_error()``. ``falcon.HTTPError`` is a
+        superclass of ``falcon.HTTPForbidden`` and a subclass of ``Exception``,
+        so it is the most specific exception type with a registered handler.
+
+        If an instance of ``falcon.HTTPNotFound`` is raised, it will be handled
+        by ``custom_handle_404()``, not by ``custom_handle_not_found()``, because
+        ``custom_handle_404()`` was registered more recently.
+
+        .. Note::
+
+            By default, the framework installs three handlers, one for
+            :class:`~.HTTPError`, one for :class:`~.HTTPStatus`, and one for
+            the standard ``Exception`` type, which prevents passing uncaught
+            exceptions to the WSGI server. These can be overridden by adding a
+            custom error handler method for the exception type in question.
+
+        Args:
+            exception (type or iterable of types): When handling a request,
+                whenever an error occurs that is an instance of the specified
+                type(s), the associated handler will be called. Either a single
+                type or an iterable of types may be specified.
+            handler (callable): A coroutine function taking the form
+                ``async func(req, resp, ex, params)``.
+
+                If not specified explicitly, the handler will default to
+                ``exception.handle``, where ``exception`` is the error
+                type specified above, and ``handle`` is a static method
+                (i.e., decorated with ``@staticmethod``) that accepts
+                the same params just described. For example::
+
+                    class CustomException(CustomBaseException):
+
+                        @staticmethod
+                        async def handle(req, resp, ex, params):
+                            # TODO: Log the error
+                            # Convert to an instance of falcon.HTTPError
+                            raise falcon.HTTPError(falcon.HTTP_792)
+
+                If an iterable of exception types is specified instead of
+                a single type, the handler must be explicitly specified.
+        """
+
+        if handler is None:
```

Comment:
I went ahead and cleaned up the method implementation here; trying to use super() just made things too convoluted, and there is no reason to support deprecated handler signatures in the new ASGI implementation.